// _@bytemd_plugin-gfm@1.17.4@@bytemd/plugin-gfm/dist/index.mjs
import * as icons from "@icon-park/svg";
import remarkGfm from "remark-gfm";
const strike = "Strikethrough";
const strikeText = "text";
const table = "Table";
const tableHeading = "Heading";
const task = "Task list";
const taskText = "todo";
const en = {
  strike,
  strikeText,
  table,
  tableHeading,
  task,
  taskText
};
function gfm({
  locale: _locale,
  ...remarkGfmOptions
} = {}) {
  const locale = { ...en, ..._locale };
  return {
    remark: (processor) => processor.use(remarkGfm, remarkGfmOptions),
    actions: [
      {
        title: locale.strike,
        icon: icons.Strikethrough({}),
        cheatsheet: `~~${locale.strikeText}~~`,
        handler: {
          type: "action",
          click({ wrapText, editor }) {
            wrapText("~~");
            editor.focus();
          }
        }
      },
      {
        title: locale.task,
        icon: icons.CheckCorrect({}),
        cheatsheet: `- [ ] ${locale.taskText}`,
        handler: {
          type: "action",
          click({ replaceLines, editor }) {
            replaceLines((line) => "- [ ] " + line);
            editor.focus();
          }
        }
      },
      {
        title: locale.table,
        icon: icons.InsertTable({}),
        handler: {
          type: "action",
          click({ editor, appendBlock, codemirror }) {
            const { line } = appendBlock(
              `| ${locale.tableHeading} |  |
| --- | --- |
|  |  |
`
            );
            editor.setSelection(
              codemirror.Pos(line, 2),
              codemirror.Pos(line, 2 + locale.tableHeading.length)
            );
            editor.focus();
          }
        }
      }
    ]
  };
}
export {
  gfm as default
};
